pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -480,13 +481,13 @@ def _validate_monitor_key(self, trainer):
             )
             raise MisconfigurationException(m)
 
-    def _get_metric_interpolated_filepath_name(self, epoch, ckpt_name_metrics):
-        filepath = self.format_checkpoint_name(epoch, ckpt_name_metrics)
+    def _get_metric_interpolated_filepath_name(self, ckpt_name_metrics: Dict[str, Any]):
+        epoch = ckpt_name_metrics.get("epoch")
+        step = ckpt_name_metrics.get("step")
```

Comment:
these keys should always exist in `ckpt_name_metrics` right? any reason not to do this? otherwise we could silently pass `None` for `epoch` or `step`

```suggestion
        epoch = ckpt_name_metrics["epoch"]
        step = ckpt_name_metrics.get["step"]
```

Reply:
not necessary for legacy checkpoints...